common/fs/src/tail.rs

```diff
@@ -87,6 +88,199 @@ impl Tailer {
         None
     }
 
+    async fn get_initial_offset(
+        target: &Path,
+        fs: &FileSystem,
+        initial_offsets: Option<HashMap<FileId, u64>>,
+        lookback_config: Lookback,
+    ) -> Option<(EntryKey, u64)> {
+        let entry_key = fs.lookup(target, &fs.entries.borrow())?;
+        let entries = fs.entries.borrow();
+        let entry = &entries.get(entry_key)?;
+        let path = fs.resolve_direct_path(entry, &fs.entries.borrow());
+        if let Entry::File { data, .. } = entry {
+            let inode: FileId = { (&data.borrow().deref().get_inode().await).into() };
+            Some((
+                entry_key,
+                match lookback_config {
+                    Lookback::Start => match initial_offsets.as_ref() {
+                        Some(initial_offsets) => {
+                            let offset = initial_offsets.get(&inode).copied().unwrap_or(0);
+                            debug!("Got offset {} from state using key {:?}", offset, path);
+                            offset
```

Comment:
We can extract the duplicate code into a `fn` or move it above, as it's not dependent on the lookback value.